### ISSUE TITLE
Fit fabric.ActiveSelection to clipped object

### ISSUE THREAD

**Issue by DefeninMaxime** [fabricjs/fabric.js#5908]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when I use `clipTo` method on a fabric.Object, the selection has been make from original content of the object and not from the real displayed content.

**Describe the solution you'd like**
Add a mode to fit the selection of a clipped object to the real displayed content instead of the original content.
The selection can be a rect, it's not necessary to follow pixel per pixel.

**Comment by asturur** [comment]
That feature is not available yet.
Clipped objects retains the object bbox.
Exception made for images, that can have a different area if used with cropX, cropY, width height

**Comment by DefeninMaxime** [comment]
I made a cropper which can rotate/skew/scale it self and placed wherever on Image. clipTo function of Image use matrix transformation of cropper object to render Image. So only cropX, cropY, width, height is not enought for me. 

Have you any idea about this feature in your roadmap ?

**Comment by asturur** [comment]
well i m not sure how your cropper works.
The idea is to move the controls rendering in the upper canvas, the function to draw the bounding box and the controls are being modified, the one for the controls are ready but i cannot release them yet.

Once that is done ( and i m using it, is not parked in a branch ) you can define control position per object, with a function, so you would be able to apply custom logic to controls.

I would suggest you to use clipPath for that, that would make easy to track clipping area position.

Does clipPath fit your use case? or the blurrines for now is stopping you?

**Comment by ShaMan123** [comment]
#7868
